sdk/state/open.go

```diff
@@ -100,7 +100,31 @@ func (c *Channel) ProposeOpen(p OpenParams) (Open, error) {
 //
 // If after confirming the open has all the signatures it needs to be fully and
 // completely signed, fully signed will be true, otherwise it will be false.
-func (c *Channel) ConfirmOpen(m Open) (open Open, authorized bool, err error) {
+func (c *Channel) ConfirmOpen(m OpenAgreement) (open OpenAgreement, authorized bool, err error) {
+	// at the end of this method, if no error, then save a new channel openAgreement. Use the
+	// channel's saved open agreement details if present, to prevent other party from changing.
+	defer func() {
+		if err != nil {
+			return
+		}
+		asset := m.Asset
+		assetLimit := m.AssetLimit
+		if c.openAgreement.Asset != nil {
+			asset = c.openAgreement.Asset
+		}
+		if c.openAgreement.AssetLimit != 0 {
+			assetLimit = c.openAgreement.AssetLimit
+		}
```

Comment:
❓ It looks like this has the values in `c` override the values in `m`, but throughout the rest of the functions we use the value for `m`, like at line 130. We should never allow us to get to the point where there are different values, so this selection shouldn't be possible, and we should always error if an open is attempted to be confirmed.

Reply:
good catch. I added an `OpenAgreementDetails` similar to the `CloseAgreementDetails`, and then validate the input in `ConfirmOpen` so it's safe to always pull from `m.Details`
https://github.com/stellar/experimental-payment-channels/pull/125/commits/f4dcd5d2b54aad8c753ca70b34b1be11a9bff40a